src/Services/Implementations/DefaultAlgoliaTaskLogger.cs

```diff
@@ -51,11 +50,37 @@ public void HandleEvent(TreeNode node, string eventName)
             }
         }
 
+        public void HandleEventAfter(TreeNode node, string eventName)
+        {
+            var treeProvider = new TreeProvider();
+            if (eventName != DocumentEvents.Delete.Name)
+            {
+                node = treeProvider.SelectSingleDocument(node.DocumentID, coupledData: true);
+            }
+            
+            foreach (var indexName in IndexStore.Instance.GetAll().Select(index => index.IndexName))
```

Comment:
Entire loop content is equal to HandleEvent method, extract it to separate private method.
Please see the comment below.

Reply:
I have extracted this loop into a private method.

Reply:
However now there is a ForceHandleEvent which only calls this private method. Because I think we should keep the names of this private method and the exposed method different.